core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -30,6 +30,7 @@ private CatalogProperties() {}
   public static final String TABLE_DEFAULT_PREFIX = "table-default.";
   public static final String TABLE_OVERRIDE_PREFIX = "table-override.";
   public static final String METRICS_REPORTER_IMPL = "metrics-reporter-impl";
+  public static final String DEFAULT_REST_FILE_IO_IMPL = "org.apache.iceberg.io.ResolvingFileIO";
```

Comment:
Why is this public? It is specific to the REST catalog so it doesn't make sense to me that it is a publicly accessible string here.